File: 백준/Silver/25757. 임스와 함께하는 미니게임/임스와 함께하는 미니게임.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());
        HashSet<String> set = new HashSet<>();
        int n =Integer.parseInt(st.nextToken());
        String game = st.nextToken();
        int cnt=0;
        int answer=0;
        switch(game) {
        case "Y":
        	for(int i=0;i<n;i++) {
        		String tmp = br.readLine();
        		if(!set.contains(tmp)) {
        			set.add(tmp);
        			cnt++;
        			if(cnt==1) {
        				cnt=0;
        				answer++;
        			}
        		}
        		
        	}
        	break;
        case "F":
        	
        	for(int i=0;i<n;i++) {
        		String tmp = br.readLine();
        		if(!set.contains(tmp)) {
        			set.add(tmp);
        			cnt++;
        			if(cnt==2) {
        				cnt=0;
        				answer++;
        			}
        		}
        		
        	}
        	break;
        case "O":
        	for(int i=0;i<n;i++) {
        		String tmp = br.readLine();
        		if(!set.contains(tmp)) {
        			set.add(tmp);
        			cnt++;
        			if(cnt==3) {
        				cnt=0;
        				answer++;
        			}
        		}
        		
        	}
        	break;
        }
        bw.write(answer+"");
        bw.close();
        br.close();
        
    }
}

File: 백준/Silver/25757. 임스와 함께하는 미니게임/임스와 함께하는 미니게임.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val input = br.readLine().split(" ")
    val n = input[0].toInt()
    val game = input[1]
    val set = HashSet<String>()
    var cnt = 0
    var answer = 0
    
    when (game) {
        "Y" -> {
            for (i in 0 until n) {
                val tmp = br.readLine()
                if (!set.contains(tmp)) {
                    set.add(tmp)
                    cnt++
                    if (cnt == 1) {
                        cnt = 0
                        answer++
                    }
                }
            }
        }
        "F" -> {
            for (i in 0 until n) {
                val tmp = br.readLine()
                if (!set.contains(tmp)) {
                    set.add(tmp)
                    cnt++
                    if (cnt == 2) {
                        cnt = 0
                        answer++
                    }
                }
            }
        }
        "O" -> {
            for (i in 0 until n) {
                val tmp = br.readLine()
                if (!set.contains(tmp)) {
                    set.add(tmp)
                    cnt++
                    if (cnt == 3) {
                        cnt = 0
                        answer++
                    }
                }
            }
        }
    }
    bw.write("$answer")
    bw.close()
    br.close()
}